matrix-sdk-crypto/src/main/java/org/matrix/androidsdk/crypto/internal/otk/OneTimeKeysResponseHandler.kt

```diff
@@ -116,4 +117,50 @@ class OneTimeKeysResponseHandler(
             }
         }
     }
+
+    fun onNetworkError(e: Exception, usersDevicesToClaim: MXUsersDevicesMap<String>) {
+        onError(usersDevicesToClaim) {
+            it.callback.onNetworkError(e)
+        }
+    }
+
+    fun onMatrixError(e: MatrixError, usersDevicesToClaim: MXUsersDevicesMap<String>) {
+        onError(usersDevicesToClaim) {
+            it.callback.onMatrixError(e)
+        }
+    }
+
+    fun onUnexpectedError(e: Exception, usersDevicesToClaim: MXUsersDevicesMap<String>) {
+        onError(usersDevicesToClaim) {
+            it.callback.onUnexpectedError(e)
+        }
+    }
+
+    private fun onError(usersDevicesToClaim: MXUsersDevicesMap<String>, callCallback: ((PendingRequest) -> Unit)) {
+        // Spread the failure to all the pending requests which are waiting for this session
+        val concernedPendingRequests = mutableSetOf<PendingRequest>()
+
+        pendingRequests.toList().forEach { pendingRequest ->
```

Comment:
Are you sur `mainForEach@` is not after `pendingRequests.toList().forEach` (line 143 instead of 144)?

Reply:
We have 3 nested `forEach`, that's why I used a label `mainForEach` for the second one, to be able to return form it when a `pendingRequest` is found in the `pendingRequests` Collection.
But we still continue to iterate on `pendingRequests`.

Reply:
Maybe the name with `main` is confusing then...

Reply:
sorry, but after thinking, I still think we should go to the next pendingRequest by jumping to the first forEach. I will need to sync irl